### PR TITLE
schema: Update the json file

### LINKED ISSUE #618: merge "ManifestDescriptor" and "Descriptor" types
Currently, we have `ManifestDescriptor` and `Descriptor` types. The only difference is the inclusion of the platform field. For making generic processors that filter based on platform type, this creates a number of complexities (see https://github.com/docker/containerd/pull/638 for an example of this pattern).

We'll simply need to merge these types in the specification and struct definitions.

### PR DESCRIPTION
**Pull request by q384566678**
Update the corresponding json file and spec file according to #620 

Signed-off-by: zhouhao <zhouhao@cn.fujitsu.com>

Fixes #618

### CODE REVIEW

File: image-index.md [hunk 1]
```diff
@@ -43,32 +43,6 @@ For the media type(s) that this document is compatible with, see the [matrix][ma
     This OPTIONAL property describes the platform which the image in the manifest runs on.
```
Comment (jonboulle) [inline image-index.md]: line 28 should be changed from "with the following additional properties and restrictions" to "with the following requirements and restrictions" or similar

**Comment by stevvooe** [comment]
Platform shouldn't be valid on descriptors outside usage on the index. While we've changed the Go datatype to make processing easier, the specification still reads correctly.

**Comment by stevvooe** [comment]
I missed https://github.com/opencontainers/image-spec/pull/638#pullrequestreview-31021778.

> Perhaps it would make more sense to leave platform where it is, and instead simply augment the descriptor description here to mention that "descriptors pointing to certain types may include extended fields" or similar ..?

This is probably the correct approach here.

**Comment by qianzhangxa** [comment]
Agree, that should be the correct approach.

File: image-index.md (hunk 1 above)
Comment (q384566678) [inline image-index.md]: update, what do you think of https://github.com/opencontainers/image-spec/pull/638/commits/3e7b8d00a1c7b5c8561c1fbcbdc8a437f6c82850

**Comment by stevvooe** [comment]
I'm going to go ahead and close this favor of @jonboulle's suggestion in https://github.com/opencontainers/image-spec/pull/638#pullrequestreview-31021778.

**Comment by qianzhangxa** [comment]
But based on @jonboulle's comment, we need to augment the descriptor description, so we still need another PR to update `descriptor.md`?

**Comment by stevvooe** [comment]
@qianzhangxa Yes, but this PR needs to be closed.

**Pull request closed without merging**